feature/deps.go

```diff
@@ -56,7 +56,7 @@ var deps = depDesc{
 		lnwire.PaymentAddrOptional: {},
 	},
 	lnwire.AnchorsOptional: {
-		lnwire.StaticRemoteKeyOptional: {},
+		lnwire.StaticRemoteKeyRequired: {},
```

Comment:
iirc this value can remain `StaticRemoteKeyOptional`. internally we end up mapping required -> optional, so we only end up checking that the dependency is _at least_ optional